src/workerd/api/tests/js-rpc-test.js

```diff
@@ -832,6 +860,106 @@ export let serializeRpcPromiseOrProprety = {
   },
 }
 
+export let streams = {
+  async test(controller, env, ctx) {
+    // Send WritableStream.
+    {
+      let { readable, writable } = new IdentityTransformStream();
+      let promise = env.MyService.writeToStream(writable);
+      let text = await new Response(readable).text();
+      assert.strictEqual(text, "foo, bar, baz!");
+      await promise;
+    }
+
+    // TODO(someday): Test JS-backed WritableStream, when it actually works.
+
+    // TODO(someday): Is there any way to construct an encoded WritableStream? Only system
+    //   streams can be encoded, but there's no API that returns an encoded WritableStream I think.
```

Comment:
What do you mean by "encoded WritableStream"?

Reply:
I mean a `WritableStream` which wraps a `WritableStreamSink` which is `EncodedAsyncOutputStream` with a `StreamEncoding` other than `IDENTITY`.

The only place I know of where our API even returns a `WritableStream` is when constructing a transform stream and those never wrap `EncodedAsyncOuputStream`.

Reply:
Ah, I think your test would have to construct it. We do have `CompressionStream` which would make for a good test, but not quite what you're asking for here.